{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "f7fb3cd8-d7e3-4cc6-b1bc-36f4f496882f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pyfish import fish_plot, process_data, setup_figure\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "\n",
    "sns.set_style('ticks', rc={\"axes.facecolor\": (0, 0, 0, 0)})\n",
    "sns.set_context('talk')\n",
    "\n",
    "from matplotlib import rcParams\n",
    "rcParams['font.family'] = 'sans-serif'\n",
    "rcParams['font.sans-serif'] = ['Arial']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "b2a0e2c6-e8b9-4443-8563-2edd39ab9e58",
   "metadata": {},
   "outputs": [],
   "source": [
    "ec_sra_path = \"../../data/ec_filtered_d_sra.tsv\"\n",
    "kp_sra_path = \"../../data/kp_filtered_d_sra.tsv\"\n",
    "pa_sra_path = \"../../data/pa_filtered_d_sra.tsv\"\n",
    "mhh_path = \"../../data/StudyDataGalardini_pseudonymized_v1_SENT_2022-11-17.xlsx\"\n",
    "cph_path = \"../../data/metadata_marco_04082023.xlsx\"\n",
    "cph_key_path = \"../../data/material_key.xlsx\"\n",
    "bs_path = \"../../data/KBS_strains+TC_Stand30052023new.xlsx\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "6e5a0550-7a55-4095-8ce8-62785ed7577b",
   "metadata": {},
   "outputs": [],
   "source": [
    "ec_sra = pd.read_csv(ec_sra_path, sep=\"\\t\")\n",
    "kp_sra = pd.read_csv(kp_sra_path, sep=\"\\t\")\n",
    "pa_sra = pd.read_csv(pa_sra_path, sep=\"\\t\")\n",
    "mhh = pd.read_excel(mhh_path)\n",
    "cph = pd.read_excel(cph_path)\n",
    "cph_k = pd.read_excel(cph_key_path)\n",
    "bs = pd.read_excel(bs_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "07a90af6-e69a-480e-a22b-e59453c28782",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "to_csv() got an unexpected keyword argument 'delimiter'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_8474/300440743.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m    131\u001b[0m \u001b[0msra\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconcat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mec_sra\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkp_sra\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpa_sra\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mignore_index\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    132\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 133\u001b[0;31m \u001b[0msra\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"../../data/sra_all.tsv\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdelimiter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"\\t\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: to_csv() got an unexpected keyword argument 'delimiter'"
     ]
    }
   ],
   "source": [
    "# get sra df\n",
    "# drop columns with missing values or useless\n",
    "ec_sra.drop(columns=[\"Month\", \"BioSampleModel\", \"geo_loc_name\", \"Strain\", \"Library Name\", \"version\", \n",
    "                      \"LibrarySelection\", \"Organism\", \"Bytes\", \"AvgSpotLen\", \"Bases\", \"DATASTORE filetype\",\n",
    "                      \"DATASTORE provider\", \"DATASTORE region\", \"Experiment\", \"LibrarySelection\", \"ReleaseDate\",\n",
    "                      \"Sample Name\", \"create_date\", \"Date\"], inplace=True)\n",
    "kp_sra.drop(columns=[\"Month\", \"Library Name\", \"version\", \"LibrarySelection\", \"Organism\", \"Bytes\", \"AvgSpotLen\", \n",
    "                     \"Bases\", \"DATASTORE filetype\", \"DATASTORE provider\", \"DATASTORE region\", \"Experiment\", \n",
    "                     \"LibrarySelection\", \"ReleaseDate\", \"Sample Name\", \"create_date\", \"Date\"], inplace=True)\n",
    "pa_sra.drop(columns=[\"Month\", \"BioSampleModel\", \"geo_loc_name\", \"Library Name\", \"version\", \n",
    "                      \"LibrarySelection\", \"Organism\", \"Bytes\", \"AvgSpotLen\", \"Bases\", \"DATASTORE filetype\",\n",
    "                      \"DATASTORE provider\", \"DATASTORE region\", \"Experiment\", \"LibrarySelection\", \"ReleaseDate\",\n",
    "                      \"Sample Name\", \"create_date\", \"BioSampleModel\", \"Date\"], inplace=True)\n",
    "\n",
    "# drop rows with assay type == ['WGA', 'CLONE', 'FINISHING', 'WCS']\n",
    "assay_type = ['WGA', 'CLONE', 'FINISHING', 'WCS']\n",
    "ec_sra = ec_sra[~ec_sra['Assay Type'].isin(assay_type)]\n",
    "kp_sra = kp_sra[~kp_sra['Assay Type'].isin(assay_type)]\n",
    "pa_sra = pa_sra[~pa_sra['Assay Type'].isin(assay_type)]\n",
    "\n",
    "# drop rows with platform == OXFORD_NANOPORE\n",
    "platform = ['OXFORD_NANOPORE']\n",
    "ec_sra = ec_sra[~ec_sra['Platform'].isin(platform)]\n",
    "kp_sra = kp_sra[~kp_sra['Platform'].isin(platform)]\n",
    "pa_sra = pa_sra[~pa_sra['Platform'].isin(platform)]\n",
    "\n",
    "# drop rows with LibraryLayout == SINGLE\n",
    "layout=['SINGLE']\n",
    "ec_sra = ec_sra[~ec_sra['LibraryLayout'].isin(layout)]\n",
    "kp_sra = kp_sra[~kp_sra['LibraryLayout'].isin(layout)]\n",
    "pa_sra = pa_sra[~pa_sra['LibraryLayout'].isin(layout)]\n",
    "\n",
    "# drop rows with LibrarySource == ['OTHER', 'METAGENOMIC', 'GENOMIC SINGLE CELL']\n",
    "l_source = ['OTHER', 'METAGENOMIC', 'GENOMIC SINGLE CELL']\n",
    "ec_sra = ec_sra[~ec_sra['LibrarySource'].isin(l_source)]\n",
    "kp_sra = kp_sra[~kp_sra['LibrarySource'].isin(l_source)]\n",
    "pa_sra = pa_sra[~pa_sra['LibrarySource'].isin(l_source)]\n",
    "\n",
    "# look at Isolation_source column - might be not human\n",
    "human_source = [\"stool\", \"feces\", \"bile\", \"urine\", \"Rectal Swab\", \"liver\", \"Femur\", \"bone marrow\", \"spleen\", \"human\",\n",
    "                \"Peritoneum\", \"Pericardia\", \"sputum\", \"fecal sample\", \"Rectal swab\", \"bloodstream\", \"Rectal abscess\",\n",
    "                \"blood\", \"Bile Duct Fluid\", \"Wound culture\", \"Rectal Specimen\", \"recto anal mucosal swab\", \"Patient Blood\",\n",
    "                \"Patient Urine\", \"Patient Sputum\", \"Patient hydatid fluid\", \"Patient Drainage\", \"Pus\", \"clinical sample\",\n",
    "                \"respiratory\", \"wound\", \"blood culture\", \"E. coli K-12 from a stool sample of a diphtheria patient in Palo Alto\\, CA in 1922\",\n",
    "                \"clinical\", \"oral\", \"puncture fluid\", \"ascitic fluid\", \"intestine\", \"throat\", \"wound swab\", \"endocervical\",\n",
    "                \"cerebrospinal fluid\", \"Abscess\", \"Peritoneal fluid\", \"CSF\", \"Ascitic fluid\", \"Abscess fluid\", \"rectal faeces\",\n",
    "                \"rectal swab\", \"CF Sputum\", \"rectal swap\", \"Necrotic tissue\", \"Blood culture\", \"Human\\, Stool\", \"Human\\, Blood culture\",\n",
    "                \"pus\", \"fluid\", \"Fluid\", \"lung\", \"bladder epithelial biopsy from 60-year-old female patient with recurrent urinary tract infection\",\n",
    "                \"bladder epithelial biopsy from 30-year-old female patient with recurrent urinary tract infection\",\n",
    "                \"bladder epithelial biopsy from 50-year-old female patient with recurrent urinary tract infection\", \"tissue\",\n",
    "                \"swab\", \"cecal content\", \"Skin swab\", \"Human gut\", \"pancreas\", \"ear\", \"peritoneal fluid\", \"vagina\", \"skin\",\n",
    "                \"stool sample\", \"lung swab\", \"uterine swab\", \"sinus swab\", \"urine swab\", \"soft tissue\", \"kidney\", \"joint swab\",\n",
    "                \"oral swab\", \"eye\", \"endometrial swab\", \"pharyngeal tissue\", \"abdomen\", \"respiratory tract\", \"Anal gland\",\n",
    "                \"Right Ear Swab\", \"Foot Swab\", \"Prostrate\", \"Wound/Abscess\", \"rectum\", \"Faecal samples\", \"Abdominal wound discharge\",\n",
    "                \"Secretion\", \"Catheter\", \"Human Sample\", \"Faeces or rectal swab\", \"Umbilical swab\", \"Wound Swab\", \"Pooled: Liver\\, Peritoneum\\, Abscess\",\n",
    "                \"nasal swab\", \"Uterine swab\", \"fecal microbiome\", \"heart swab\", \"liver swab\", \"vaginal swab\", \"uterus\", \"joint fluid\",\n",
    "                \"pericardium\", \"umbilicus\", \"lungs\", \"outbreaks of food poisoning\", \"diarrhea from outbreaks of food poisoning\",\n",
    "                \"Uterus\", \"semen\", \"nasal cavity\", \"rectal feces\", \"Blood_Blood\", \"swab (superficial and non-surgical)\", \"swab (cavity)\",\n",
    "                \"unspecified diagnostic sample\", \"swab (wound)\", \"punctate\", \"biopsy\", \"surgical sample\", \"PUS\", \"Pus from Wound\",\n",
    "                \"Tissue\", \"Blood specimen\", \"Small intestine\", \"Urine_\", \"Exudate\", \"Catheter Tip\", \"Anal glands\", \"Skin wound\",\n",
    "                \"Abdominal\", \"Perirectal\", \"FLUID\", \"TISSUE\", \"ICU staff rectal swab\", \"blood draw same patient as MB9267\", \"Peritoneal Fluid\",\n",
    "                \"Wound Discharge\", \"Cecal Contents\", \"Diseased Organ\", \"Perianal Swab\", \"Perineum Swab\", \"Rectal Isolate\",\n",
    "                \"Abdomen Fluid\", \"Abdomen Tissue\", \"Kidney Swab\", \"Perineal Swab\", \"Endotracheal Aspirate\", \"Perianal\", \"Catheter Urine\",\n",
    "                \"rectal\", \"Blood Specimen\", \"Cecum\", \"Faecal swab\", \"hand swab (farm workers)\", \"feces (farm workers)\",\n",
    "                \"cecal\", \"nose swab (farm workers)\", \"nose swab (abattoir operator households)\", \"anal swab\", \"hand swab (abattoir operator households)\",\n",
    "                \"feces (abattoir operator households)\", \"abdominal drainage\", \"midstream urine\", \"Feaces\", \"Perianal Abscess Swab\",\n",
    "                \"Mid stream Urine\", \"Anal Swab\", \"Aortic Tissue\", \"Screening Swab\", \"Ankle Wound Tissue\", \"Periph.Ins.Central Catheter Tip\",\n",
    "                \"Cecal content\", \"secretion\"]\n",
    "hospital_source = [\"hospital sewage\", \"hospital\", \"Hospital influent\", \"Hospital efluent\", \"Baby Bath Drain Special Care Nursery\"]\n",
    "environmental_source = [\"food\", \"beef trim\", \"ground beed patty\", \"Washington\", \"soil\", \"untreated waste water\", \n",
    "                      \"potable source water\", \"pizza dough dry mixture\", \"Raw Beef\", \"Apple cider\", \"flour\",\n",
    "                      \"venison\", \"lettuce\", \"spinach\", \"environmental\", \"romaine lettuce\", \"leafy green\",\n",
    "                      \"waste water primary treatment\", \"Wastewater\\, primary treatment\", \"environmental reclaimed storage\",\n",
    "                      \"kale\", \"drag swab through chicken house (Gallus gallus domesticus)\", \"milk\",\n",
    "                      \"all-purpose wheat flour\", \"water\", \"cloaca\", \"chicken house back\", \"chicken house middle\",\n",
    "                      \"chicken house front\", \"Sewage water\", \"pork\", \"water sewage\", \"environment\", \"environmental sample\",\n",
    "                      \"river water\", \"sediment\", \"household environment\", \"surface\", \"doorknob\", \"control air\", \"reservoir water\",\n",
    "                      \"Environment\", \"water reservoir\", \"Raw wastewater from aircraft arriving at an international German airport\",\n",
    "                      \"Raw wastewater from three airplanes arriving from Singapore\\, Mombasa and Canada\", \"cattle farm waste water\",\n",
    "                      \"wastewater\", \"environmental swab lairage\", \"Environmental sample: soil sample\", \"cloacal swab\",\n",
    "                      \"environmental swab\", \"Cistern water\", \"pond water\", \"water cistern\", \"fresh water stream surface\",\n",
    "                      \"Manawatu river\", \"cattle farm\", \"dairy farm field soil\", \"Storm water drain\", \"Floor swab\", \"sold in market\",\n",
    "                      \"Domestic Wastewaters\", \"Barn Environment\", \"Water from irrigation pond\", \"water pond\", \"water (slaughter house)\",\n",
    "                      \"water (farm)\", \"lake water\", \"Soil/Sediment\", \"marine sediment\"] # including food, wastewatter\n",
    "\n",
    "ec_sra[\"isolation_source_categ\"] = [\"human\" if x in human_source\n",
    "                                    else \"hospital\" if x in hospital_source\n",
    "                                    else \"evironmental\" if x in environmental_source\n",
    "                                    else \"other\"\n",
    "                                    for x in ec_sra[\"Isolation_source\"]]\n",
    "kp_sra[\"isolation_source_categ\"] = [\"human\" if x in human_source\n",
    "                                    else \"hospital\" if x in hospital_source\n",
    "                                    else \"evironmental\" if x in environmental_source\n",
    "                                    else \"other\"\n",
    "                                    for x in kp_sra[\"Isolation_source\"]]\n",
    "pa_sra[\"isolation_source_categ\"] = [\"human\" if x in human_source\n",
    "                                    else \"hospital\" if x in hospital_source\n",
    "                                    else \"evironmental\" if x in environmental_source\n",
    "                                    else \"other\"\n",
    "                                    for x in pa_sra[\"Isolation_source\"]]\n",
    "\n",
    "# Keep Consent column, to add private in the hospital data\n",
    "\n",
    "# Collection column as new, to add if its public, mhh, cph, bs\n",
    "ec_sra[\"collection\"] = ['SRA' if x != 0\n",
    "                       else 'SRA'\n",
    "                       for x in ec_sra[\"Run\"]]\n",
    "kp_sra[\"collection\"] = ['SRA' if x != 0\n",
    "                       else 'SRA'\n",
    "                       for x in kp_sra[\"Run\"]]\n",
    "pa_sra[\"collection\"] = ['SRA' if x != 0\n",
    "                       else 'SRA'\n",
    "                       for x in pa_sra[\"Run\"]]\n",
    "\n",
    "# rename geo_loc_name_country and geo_loc_name_country_continent for country and continent\n",
    "ec_sra.rename(columns={\"geo_loc_name_country\":\"country\", \"geo_loc_name_country_continent\":\"continent\"})\n",
    "kp_sra.rename(columns={\"geo_loc_name_country\":\"country\", \"geo_loc_name_country_continent\":\"continent\"})\n",
    "pa_sra.rename(columns={\"geo_loc_name_country\":\"country\", \"geo_loc_name_country_continent\":\"continent\"})\n",
    "\n",
    "# add sp column\n",
    "ec_sra[\"sp\"] = ['ecoli' if x != 0\n",
    "                       else 'ecoli'\n",
    "                       for x in ec_sra[\"Run\"]]\n",
    "kp_sra[\"sp\"] = ['kpneumoniae' if x != 0\n",
    "                       else 'kpneumoniae'\n",
    "                       for x in kp_sra[\"Run\"]]\n",
    "pa_sra[\"sp\"] = ['paeruginosa' if x != 0\n",
    "                       else 'paeruginosa'\n",
    "                       for x in pa_sra[\"Run\"]]\n",
    "\n",
    "sra = pd.concat([ec_sra, kp_sra, pa_sra], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "22a95677-7843-4170-acfa-8bbdf9f6bfc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "sra.to_csv(\"../../data/sra_all.tsv\", sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "d0f332ce-56a9-4471-a4fc-e51c1b20d98e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>BioProject</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Center Name</th>\n",
       "      <th>Consent</th>\n",
       "      <th>Instrument</th>\n",
       "      <th>LibraryLayout</th>\n",
       "      <th>LibrarySource</th>\n",
       "      <th>Platform</th>\n",
       "      <th>SRA Study</th>\n",
       "      <th>geo_loc_name_country</th>\n",
       "      <th>geo_loc_name_country_continent</th>\n",
       "      <th>Isolation_source</th>\n",
       "      <th>Collection_Date</th>\n",
       "      <th>Year</th>\n",
       "      <th>isolation_source_categ</th>\n",
       "      <th>collection</th>\n",
       "      <th>sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR2910655</td>\n",
       "      <td>WGS</td>\n",
       "      <td>PRJNA298331</td>\n",
       "      <td>SAMN04191544</td>\n",
       "      <td>CFSAN</td>\n",
       "      <td>public</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>SRP064902</td>\n",
       "      <td>USA</td>\n",
       "      <td>North America</td>\n",
       "      <td>fecal beef heifer</td>\n",
       "      <td>2015-04-16</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>other</td>\n",
       "      <td>SRA</td>\n",
       "      <td>ecoli</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR2035435</td>\n",
       "      <td>WGS</td>\n",
       "      <td>PRJNA230969</td>\n",
       "      <td>SAMN03581169</td>\n",
       "      <td>CFSAN</td>\n",
       "      <td>public</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>SRP058582</td>\n",
       "      <td>USA</td>\n",
       "      <td>North America</td>\n",
       "      <td>food</td>\n",
       "      <td>2015-04-16</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>evironmental</td>\n",
       "      <td>SRA</td>\n",
       "      <td>ecoli</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR2035445</td>\n",
       "      <td>WGS</td>\n",
       "      <td>PRJNA230969</td>\n",
       "      <td>SAMN03581170</td>\n",
       "      <td>CFSAN</td>\n",
       "      <td>public</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>SRP058582</td>\n",
       "      <td>USA</td>\n",
       "      <td>North America</td>\n",
       "      <td>food</td>\n",
       "      <td>2015-04-16</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>evironmental</td>\n",
       "      <td>SRA</td>\n",
       "      <td>ecoli</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR2136798</td>\n",
       "      <td>WGS</td>\n",
       "      <td>PRJNA284275</td>\n",
       "      <td>SAMN03892147</td>\n",
       "      <td>CFSAN</td>\n",
       "      <td>public</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>SRP061878</td>\n",
       "      <td>USA</td>\n",
       "      <td>North America</td>\n",
       "      <td>beef trim</td>\n",
       "      <td>2015-06-03</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>evironmental</td>\n",
       "      <td>SRA</td>\n",
       "      <td>ecoli</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR2121689</td>\n",
       "      <td>WGS</td>\n",
       "      <td>PRJNA243331</td>\n",
       "      <td>SAMN03580930</td>\n",
       "      <td>CFSAN</td>\n",
       "      <td>public</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>SRP042343</td>\n",
       "      <td>USA</td>\n",
       "      <td>North America</td>\n",
       "      <td>ground beef patty</td>\n",
       "      <td>2015-04-14</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>other</td>\n",
       "      <td>SRA</td>\n",
       "      <td>ecoli</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42035</th>\n",
       "      <td>SRR23456569</td>\n",
       "      <td>WGS</td>\n",
       "      <td>PRJNA934930</td>\n",
       "      <td>SAMN33288620</td>\n",
       "      <td>CEDARS SINAI MEDICAL CENTER</td>\n",
       "      <td>public</td>\n",
       "      <td>NextSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>SRP422491</td>\n",
       "      <td>USA</td>\n",
       "      <td>North America</td>\n",
       "      <td>ICU respiratory sample</td>\n",
       "      <td>2017-02-13</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>other</td>\n",
       "      <td>SRA</td>\n",
       "      <td>paeruginosa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42036</th>\n",
       "      <td>SRR23456570</td>\n",
       "      <td>WGS</td>\n",
       "      <td>PRJNA934930</td>\n",
       "      <td>SAMN33288619</td>\n",
       "      <td>CEDARS SINAI MEDICAL CENTER</td>\n",
       "      <td>public</td>\n",
       "      <td>NextSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>SRP422491</td>\n",
       "      <td>USA</td>\n",
       "      <td>North America</td>\n",
       "      <td>ICU respiratory sample</td>\n",
       "      <td>2017-02-13</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>other</td>\n",
       "      <td>SRA</td>\n",
       "      <td>paeruginosa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42037</th>\n",
       "      <td>SRR23456571</td>\n",
       "      <td>WGS</td>\n",
       "      <td>PRJNA934930</td>\n",
       "      <td>SAMN33288618</td>\n",
       "      <td>CEDARS SINAI MEDICAL CENTER</td>\n",
       "      <td>public</td>\n",
       "      <td>NextSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>SRP422491</td>\n",
       "      <td>USA</td>\n",
       "      <td>North America</td>\n",
       "      <td>ICU respiratory sample</td>\n",
       "      <td>2018-05-17</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>other</td>\n",
       "      <td>SRA</td>\n",
       "      <td>paeruginosa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42038</th>\n",
       "      <td>SRR23456572</td>\n",
       "      <td>WGS</td>\n",
       "      <td>PRJNA934930</td>\n",
       "      <td>SAMN33288609</td>\n",
       "      <td>CEDARS SINAI MEDICAL CENTER</td>\n",
       "      <td>public</td>\n",
       "      <td>NextSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>SRP422491</td>\n",
       "      <td>USA</td>\n",
       "      <td>North America</td>\n",
       "      <td>ICU respiratory sample</td>\n",
       "      <td>2016-08-13</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>other</td>\n",
       "      <td>SRA</td>\n",
       "      <td>paeruginosa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42039</th>\n",
       "      <td>SRR23456573</td>\n",
       "      <td>WGS</td>\n",
       "      <td>PRJNA934930</td>\n",
       "      <td>SAMN33288608</td>\n",
       "      <td>CEDARS SINAI MEDICAL CENTER</td>\n",
       "      <td>public</td>\n",
       "      <td>NextSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>SRP422491</td>\n",
       "      <td>USA</td>\n",
       "      <td>North America</td>\n",
       "      <td>ICU respiratory sample</td>\n",
       "      <td>2016-08-13</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>other</td>\n",
       "      <td>SRA</td>\n",
       "      <td>paeruginosa</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42040 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Run Assay Type   BioProject     BioSample  \\\n",
       "0       SRR2910655        WGS  PRJNA298331  SAMN04191544   \n",
       "1       SRR2035435        WGS  PRJNA230969  SAMN03581169   \n",
       "2       SRR2035445        WGS  PRJNA230969  SAMN03581170   \n",
       "3       SRR2136798        WGS  PRJNA284275  SAMN03892147   \n",
       "4       SRR2121689        WGS  PRJNA243331  SAMN03580930   \n",
       "...            ...        ...          ...           ...   \n",
       "42035  SRR23456569        WGS  PRJNA934930  SAMN33288620   \n",
       "42036  SRR23456570        WGS  PRJNA934930  SAMN33288619   \n",
       "42037  SRR23456571        WGS  PRJNA934930  SAMN33288618   \n",
       "42038  SRR23456572        WGS  PRJNA934930  SAMN33288609   \n",
       "42039  SRR23456573        WGS  PRJNA934930  SAMN33288608   \n",
       "\n",
       "                       Center Name Consent      Instrument LibraryLayout  \\\n",
       "0                            CFSAN  public  Illumina MiSeq        PAIRED   \n",
       "1                            CFSAN  public  Illumina MiSeq        PAIRED   \n",
       "2                            CFSAN  public  Illumina MiSeq        PAIRED   \n",
       "3                            CFSAN  public  Illumina MiSeq        PAIRED   \n",
       "4                            CFSAN  public  Illumina MiSeq        PAIRED   \n",
       "...                            ...     ...             ...           ...   \n",
       "42035  CEDARS SINAI MEDICAL CENTER  public    NextSeq 2000        PAIRED   \n",
       "42036  CEDARS SINAI MEDICAL CENTER  public    NextSeq 2000        PAIRED   \n",
       "42037  CEDARS SINAI MEDICAL CENTER  public    NextSeq 2000        PAIRED   \n",
       "42038  CEDARS SINAI MEDICAL CENTER  public    NextSeq 2000        PAIRED   \n",
       "42039  CEDARS SINAI MEDICAL CENTER  public    NextSeq 2000        PAIRED   \n",
       "\n",
       "      LibrarySource  Platform  SRA Study geo_loc_name_country  \\\n",
       "0           GENOMIC  ILLUMINA  SRP064902                  USA   \n",
       "1           GENOMIC  ILLUMINA  SRP058582                  USA   \n",
       "2           GENOMIC  ILLUMINA  SRP058582                  USA   \n",
       "3           GENOMIC  ILLUMINA  SRP061878                  USA   \n",
       "4           GENOMIC  ILLUMINA  SRP042343                  USA   \n",
       "...             ...       ...        ...                  ...   \n",
       "42035       GENOMIC  ILLUMINA  SRP422491                  USA   \n",
       "42036       GENOMIC  ILLUMINA  SRP422491                  USA   \n",
       "42037       GENOMIC  ILLUMINA  SRP422491                  USA   \n",
       "42038       GENOMIC  ILLUMINA  SRP422491                  USA   \n",
       "42039       GENOMIC  ILLUMINA  SRP422491                  USA   \n",
       "\n",
       "      geo_loc_name_country_continent        Isolation_source Collection_Date  \\\n",
       "0                      North America       fecal beef heifer      2015-04-16   \n",
       "1                      North America                    food      2015-04-16   \n",
       "2                      North America                    food      2015-04-16   \n",
       "3                      North America               beef trim      2015-06-03   \n",
       "4                      North America       ground beef patty      2015-04-14   \n",
       "...                              ...                     ...             ...   \n",
       "42035                  North America  ICU respiratory sample      2017-02-13   \n",
       "42036                  North America  ICU respiratory sample      2017-02-13   \n",
       "42037                  North America  ICU respiratory sample      2018-05-17   \n",
       "42038                  North America  ICU respiratory sample      2016-08-13   \n",
       "42039                  North America  ICU respiratory sample      2016-08-13   \n",
       "\n",
       "         Year isolation_source_categ collection           sp  \n",
       "0      2015.0                  other        SRA        ecoli  \n",
       "1      2015.0           evironmental        SRA        ecoli  \n",
       "2      2015.0           evironmental        SRA        ecoli  \n",
       "3      2015.0           evironmental        SRA        ecoli  \n",
       "4      2015.0                  other        SRA        ecoli  \n",
       "...       ...                    ...        ...          ...  \n",
       "42035  2017.0                  other        SRA  paeruginosa  \n",
       "42036  2017.0                  other        SRA  paeruginosa  \n",
       "42037  2018.0                  other        SRA  paeruginosa  \n",
       "42038  2016.0                  other        SRA  paeruginosa  \n",
       "42039  2016.0                  other        SRA  paeruginosa  \n",
       "\n",
       "[42040 rows x 19 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sra"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3bcc4b8f-96c2-4611-8f0a-273564539b44",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "8e927186-cc4d-45db-bb77-545f98ca1399",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pseudonym</th>\n",
       "      <th>kon</th>\n",
       "      <th>alder</th>\n",
       "      <th>PRV_MAT_TXT</th>\n",
       "      <th>provedato</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SH001x01</td>\n",
       "      <td>M</td>\n",
       "      <td>74.380822</td>\n",
       "      <td>Urin fra reservoir (Bricker)</td>\n",
       "      <td>2022-08-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SH001x02</td>\n",
       "      <td>M</td>\n",
       "      <td>57.660274</td>\n",
       "      <td>Urin fra reservoir (Bricker)</td>\n",
       "      <td>2022-08-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SH001x03</td>\n",
       "      <td>M</td>\n",
       "      <td>81.638356</td>\n",
       "      <td>Urin fra KAD</td>\n",
       "      <td>2022-08-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SH001x04</td>\n",
       "      <td>K</td>\n",
       "      <td>39.863014</td>\n",
       "      <td>Blod fra kateter (kolbe)</td>\n",
       "      <td>2022-08-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SH001x05</td>\n",
       "      <td>M</td>\n",
       "      <td>61.147945</td>\n",
       "      <td>Ekspektorat</td>\n",
       "      <td>2022-08-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20154</th>\n",
       "      <td>SH300920x2x95</td>\n",
       "      <td>M</td>\n",
       "      <td>76.167123</td>\n",
       "      <td>Urin (midtstråle)</td>\n",
       "      <td>2020-06-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20155</th>\n",
       "      <td>SH300920x2x96</td>\n",
       "      <td>M</td>\n",
       "      <td>74.947945</td>\n",
       "      <td>Urin (midtstråle)</td>\n",
       "      <td>2020-06-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20156</th>\n",
       "      <td>Sh182x02</td>\n",
       "      <td>K</td>\n",
       "      <td>47.457534</td>\n",
       "      <td>Podning cicatrice</td>\n",
       "      <td>2022-07-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20157</th>\n",
       "      <td>Sh184x02</td>\n",
       "      <td>K</td>\n",
       "      <td>72.498630</td>\n",
       "      <td>Urin fra KAD</td>\n",
       "      <td>2022-07-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20158</th>\n",
       "      <td>Sh277x02</td>\n",
       "      <td>K</td>\n",
       "      <td>83.586301</td>\n",
       "      <td>Urin fra KAD</td>\n",
       "      <td>2023-04-20</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20159 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           pseudonym kon      alder                   PRV_MAT_TXT  provedato\n",
       "0           SH001x01   M  74.380822  Urin fra reservoir (Bricker) 2022-08-03\n",
       "1           SH001x02   M  57.660274  Urin fra reservoir (Bricker) 2022-08-04\n",
       "2           SH001x03   M  81.638356                  Urin fra KAD 2022-08-04\n",
       "3           SH001x04   K  39.863014      Blod fra kateter (kolbe) 2022-08-05\n",
       "4           SH001x05   M  61.147945                   Ekspektorat 2022-08-04\n",
       "...              ...  ..        ...                           ...        ...\n",
       "20154  SH300920x2x95   M  76.167123             Urin (midtstråle) 2020-06-22\n",
       "20155  SH300920x2x96   M  74.947945             Urin (midtstråle) 2020-06-23\n",
       "20156       Sh182x02   K  47.457534             Podning cicatrice 2022-07-01\n",
       "20157       Sh184x02   K  72.498630                  Urin fra KAD 2022-07-10\n",
       "20158       Sh277x02   K  83.586301                  Urin fra KAD 2023-04-20\n",
       "\n",
       "[20159 rows x 5 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cph"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
